cmd/create.go

```diff
@@ -285,56 +275,53 @@ func (cc *createCmd) generateDockerfile(langConfig *config.DraftConfig, lowerLan
 		}
 	}
 
-	var inputs map[string]string
 	if cc.createConfig.LanguageVariables == nil {
-		inputs, err = prompts.RunPromptsFromConfigWithSkips(langConfig, maps.Keys(flagVariablesMap))
-		if err != nil {
+		handleFlagVariables(flagVariablesMap, langConfig)
+
+		if err = prompts.RunPromptsFromConfigWithSkips(langConfig); err != nil {
 			return err
 		}
 	} else {
-		inputs, err = validateConfigInputsToPrompts(langConfig, cc.createConfig.LanguageVariables)
+		err = validateConfigInputsToPrompts(langConfig, cc.createConfig.LanguageVariables)
 		if err != nil {
 			return err
 		}
 	}
 
 	if cc.templateVariableRecorder != nil {
-		for k, v := range inputs {
-			cc.templateVariableRecorder.Record(k, v)
+		for _, variable := range langConfig.Variables {
+			cc.templateVariableRecorder.Record(variable.Name, variable.Value)
 		}
 	}
 
-	maps.Copy(inputs, flagVariablesMap)
-
-	if err = cc.supportedLangs.CreateDockerfileForLanguage(lowerLang, inputs, cc.templateWriter); err != nil {
+	if err = cc.supportedLangs.CreateDockerfileForLanguage(lowerLang, langConfig, cc.templateWriter); err != nil {
 		return fmt.Errorf("there was an error when creating the Dockerfile for language %s: %w", cc.createConfig.LanguageType, err)
 	}
 
 	log.Info("--> Creating Dockerfile...\n")
-	return err
+	return nil
 }
 
 func (cc *createCmd) createDeployment() error {
 	log.Info("--- Deployment File Creation ---")
 	d := deployments.CreateDeploymentsFromEmbedFS(template.Deployments, cc.dest)
 	var deployType string
-	var customInputs map[string]string
+	var deployConfig *config.DraftConfig
 	var err error
 
 	if cc.createConfig.DeployType != "" {
 		deployType = strings.ToLower(cc.createConfig.DeployType)
-		deployConfig, err := d.GetConfig(deployType)
+		deployConfig, err = d.GetConfig(deployType)
 		if err != nil {
 			return err
 		}
 		if deployConfig == nil {
 			return errors.New("invalid deployment type")
 		}
-		customInputs, err = validateConfigInputsToPrompts(deployConfig, cc.createConfig.DeployVariables)
```

Comment:
💡 Observation to share: Hope Draft key folks might take a look, I will do small bird fyi to @davidgamero for these changes. Hope this don't fall away from any initial design decisions et. al. 

There are bunch of these changes, and this is jsut as fyi for all.

Reply:
thank you for the ping @Tatsinnit these changes look good i agree we can improve our documentation to clarify that these changes center the source-of-truth around the draftConfig instead of the previous model including input maps